ciao-controller/instance.go

```diff
@@ -46,7 +46,7 @@ type instance struct {
 	startTime time.Time
 }
 
-func isCNCIWorkload(workload *types.Workload) bool {
+func isCNCIWorkload(workload types.Workload) bool {
```

Comment:
I think it's idiomatic to not use pointers in a case like this.  It's probably not really a copy under the covers, unless the function code modifies the argument's contents.  Trust in the go memory manager.